lib/coingecko/coingecko_providers.dart

```diff
@@ -1,110 +1,38 @@
-import 'dart:convert';
-
-import 'package:decimal/decimal.dart';
+import 'dart:async';
 import 'package:flutter_riverpod/flutter_riverpod.dart';
-import 'package:http/http.dart' as http;
+import 'package:kaspium_wallet/coingecko/coingecko_repository.dart';
+import 'package:decimal/decimal.dart';
 
-import '../app_providers.dart';
-import '../settings/available_currency.dart';
+import '../../../app_providers.dart';
 import 'coingecko_types.dart';
+import '../../../settings/available_currency.dart';
 
-final _kaspaPriceCacheProvider = StateProvider<CoinGeckoPrice>((ref) {
+final _coinGeckoPriceCacheProvider = StateProvider<CoinGeckoPrice>((ref) {
```

Comment:
This was written before FutureProvider had access to previous value while refreshing, so the interface would flicker/show 0 while refreshing.

Reply:
Maybe this can be changed through using the isLoading(_), onError built in provider functionality:

`
Widget build(BuildContext context, WidgetRef ref) {
         final someProvider = ref.watch(someProvider);
         return someProvider.when(
                    loading: () => //return something for loading,
                    error: (err, stack) => //display error message,
                    data: (_) => //show correct value,
        );
}`

If you are ok with it I could work on this next with extracting the number conversion code out of the views plus making it testable.

Reply:
No need in the case of kaspa price provider. We need the last known value. Anyway the kaspa price will be saved between app runs so a StateNotifierProvider might make more sense in the future. For now, it's okay like how it is.